ethereum/blockwatch/client.go

```diff
@@ -22,29 +26,64 @@ type Client interface {
 
 // RpcClient is a Client for fetching Ethereum blocks from a specific JSON-RPC endpoint.
 type RpcClient struct {
+	rpcClient      *rpc.Client
 	client         *ethclient.Client
 	requestTimeout time.Duration
 }
 
 // NewRpcClient returns a new Client for fetching Ethereum blocks using the given
 // ethclient.Client.
-func NewRpcClient(ethClient *ethclient.Client, requestTimeout time.Duration) (*RpcClient, error) {
-	return &RpcClient{ethClient, requestTimeout}, nil
+func NewRpcClient(rpcURL string, requestTimeout time.Duration) (*RpcClient, error) {
+	ethClient, err := ethclient.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	rpcClient, err := rpc.Dial(rpcURL)
+	if err != nil {
+		return nil, err
+	}
+	return &RpcClient{rpcClient, ethClient, requestTimeout}, nil
```

Comment:
We should use explicit field names here. Otherwise the compiler won't help us if we add new fields in the future.

```suggestion
	return &RpcClient{rpcClient: rpcClient, client: ethClient, requestTimeout: requestTimeout}, nil
```